MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView.swift

```diff
@@ -6,20 +6,14 @@ class CategorisedCollectionView: UIView {
     let defaultInset: CGFloat = 20
     let defaultSpacing: CGFloat = 10
 
-    var onCategoryChanged: (Int) -> Void = { _ in }
-    var onMovieFavorited: (Int) -> Void = { _ in }
-    var onMovieSelected: (Int) -> Void = { _ in }
-
+    var didSelectMovie: (Int) -> Void = { _ in }
```

Comment:
Why don't you have a PasstroughSubject and create a `var movieSelected: AnyPublisher<Int, Never>()`?